### PR TITLE
FIX: Smoke grenade trigger firenear EH

### PR DESCRIPTION
**Pull request by Vdauphin**
- FIX: Smoke grenade trigger firenear EH (@Vdauphin).

**When merged this pull request will:**
- Fix https://github.com/Vdauphin/HeartsAndMinds/issues/512

**Final test:**
- [x] local
- [ ] server

### CODE REVIEW

**Comment by 1kuemmel1** [comment]
Hiho, just an addition:
 - ACE throwing is not recognized
 - incendiary grenades and stun grenades are ignored (in BIS and ACE throwing)

**Comment by Vdauphin** [comment]
I investigated but sadly didn't find a decent work around. The main idea is to catch when people fire around civilan to ask them to stop instead of using the ACE interaction system.  From that point of view trowing grenade, smoke, incendiary grenades and stun grenades are out of the scope of this.

I agree the filter (https://github.com/Vdauphin/HeartsAndMinds/pull/513/files#diff-a90cacb336123d2a13f7e164a1543d48R3) keep grenade but strictly speaking it should not as this is dedicated for fire weapon.

> ACE throwing is not recognized

So basically let ignore this because it is out of the scope of the firenear event and simpler for the coding part. 

> incendiary grenades and stun grenades are ignored (in BIS and ACE throwing)

Also grenade and incendiary grenades will decrease reputation of player if they do damaged to the civilian or kill him so in any case player lost reputation because he is doing bad things.

Let me know if your are agree.

**Comment by 1kuemmel1** [comment]
Too bad that it is not so easy but maybe we can make a workaround again later. 

> The main idea is to catch when people fire around civilan to ask them to stop

This is a great idea 👍 


> Also grenade and incendiary grenades will decrease reputation of player if they do damaged to the civilian or kill him, so in any case player lost reputation because he is doing bad things.

On a local test, I can't confirm this (testing both ACE and BIS throwing).

**Comment by Vdauphin** [comment]
> Unfortunately, I can not confirm that with a local test (for both ACE and BIS throwing).

I disagree at least for hand grenade (not skilled enough to kill civilian with incendiary grenades x) ).
You can see here the log of the EH damaged (`REP HD`) and EH killed (`REP KILLED`) :

```
10:28:31 "fnc_rep_change = GLOBAL 175 - CHANGE -10"
10:28:31 "REP HD = GREP 165 THIS = [C Alpha 1-6:1,""body"",0.024,B Alpha 1-1:1 (Vdauphin),""GrenadeHand"",7,B Alpha 1-1:1 (Vdauphin),""HitBody""]"
10:28:31 "fnc_rep_change = GLOBAL 165 - CHANGE -10"
10:28:31 "REP KILLED = GREP 155 THIS = [C Alpha 1-6:1,B Alpha 1-1:1 (Vdauphin),B Alpha 1-1:1 (Vdauphin),true]"
```
[Arma3_x64_2018-04-04_09-35-41.txt](https://github.com/Vdauphin/HeartsAndMinds/files/1879382/Arma3_x64_2018-04-04_09-35-41.txt)

May be more investigation should be done here.
